## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 452491`
Commit: [`1eda97a42d984e168db1ee0502187432af647167`](https://github.com/NixOS/nixpkgs/commit/1eda97a42d984e168db1ee0502187432af647167) ([subsequent changes](https://github.com/NixOS/nixpkgs/compare/1eda97a42d984e168db1ee0502187432af647167..pull/452491/head))
Merge: [`44ab40d1aac42906c5904922d6a11f711c17d732`](https://github.com/NixOS/nixpkgs/commit/44ab40d1aac42906c5904922d6a11f711c17d732)

Logs: https://github.com/eljamm/nixpkgs-review-gha/actions/runs/18552690820


---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 13 packages built:</summary>
  <ul>
    <li>lenovo-legion</li>
    <li>lenovo-legion.dist</li>
    <li>linuxKernel.packages.linux_5_15.lenovo-legion-module</li>
    <li>linuxKernel.packages.linux_6_1.lenovo-legion-module</li>
    <li>linuxPackages.lenovo-legion-module (linuxKernel.packages.linux_6_12.lenovo-legion-module)</li>
    <li>linuxKernel.packages.linux_hardened.lenovo-legion-module (linuxKernel.packages.linux_6_12_hardened.lenovo-legion-module)</li>
    <li>linuxKernel.packages.linux_6_16.lenovo-legion-module</li>
    <li>linuxPackages_latest.lenovo-legion-module (linuxKernel.packages.linux_6_17.lenovo-legion-module)</li>
    <li>linuxKernel.packages.linux_6_6.lenovo-legion-module</li>
    <li>linuxPackages_lqx.lenovo-legion-module (linuxKernel.packages.linux_lqx.lenovo-legion-module)</li>
    <li>linuxPackages_xanmod.lenovo-legion-module (linuxKernel.packages.linux_xanmod.lenovo-legion-module)</li>
    <li>linuxPackages_xanmod_latest.lenovo-legion-module (linuxKernel.packages.linux_xanmod_latest.lenovo-legion-module, linuxPackages_xanmod_stable.lenovo-legion-module)</li>
    <li>linuxPackages_zen.lenovo-legion-module (linuxKernel.packages.linux_zen.lenovo-legion-module)</li>
  </ul>
</details>


---
### `aarch64-linux`
<details>
  <summary>:x: 7 packages failed to build:</summary>
  <ul>
    <li>linuxKernel.packages.linux_5_15.lenovo-legion-module</li>
    <li>linuxKernel.packages.linux_6_1.lenovo-legion-module</li>
    <li>linuxPackages.lenovo-legion-module (linuxKernel.packages.linux_6_12.lenovo-legion-module)</li>
    <li>linuxKernel.packages.linux_hardened.lenovo-legion-module (linuxKernel.packages.linux_6_12_hardened.lenovo-legion-module)</li>
    <li>linuxKernel.packages.linux_6_16.lenovo-legion-module</li>
    <li>linuxPackages_latest.lenovo-legion-module (linuxKernel.packages.linux_6_17.lenovo-legion-module)</li>
    <li>linuxKernel.packages.linux_6_6.lenovo-legion-module</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>lenovo-legion</li>
    <li>lenovo-legion.dist</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `aarch64-linux`</summary>
<details>
<summary>linuxKernel.packages.linux_5_15.lenovo-legion-module</summary>
<pre>/nix/store/6516wznka0xwdkzx0cpgfl4wmx8d5r5y-linux-5.15.194-dev/lib/modules/5.15.194/source/include/linux/printk.h:450:44: warning: &#x27;fancurve.size&#x27; is used uninitialized [8;;[https://gcc.gnu.org/onlinedocs/gcc-14.3.0/gcc/Warning-Options.html#index-Wno-uninitialized-Wuninitialized8;;]](https://gcc.gnu.org/onlinedocs/gcc-14.3.0/gcc/Warning-Options.html#index-Wno-uninitialized%07-Wuninitialized8;;%07])
  450 | #define printk(fmt, ...) printk_index_wrap(_printk, fmt, ##__VA_ARGS__)
      |                                            ^
/nix/store/6516wznka0xwdkzx0cpgfl4wmx8d5r5y-linux-5.15.194-dev/lib/modules/5.15.194/source/include/linux/printk.h:422:17: note: in definition of macro &#x27;printk_index_wrap&#x27;
  422 |                 _p_func(_fmt, ##__VA_ARGS__);                           \
      |                 ^~~~~~~
/nix/store/6516wznka0xwdkzx0cpgfl4wmx8d5r5y-linux-5.15.194-dev/lib/modules/5.15.194/source/include/linux/printk.h:523:9: note: in expansion of macro &#x27;printk&#x27;
  523 |         printk(KERN_INFO pr_fmt(fmt), ##__VA_ARGS__)
      |         ^~~~~~
/build/source/kernel_module/legion-laptop.c:5241:9: note: in expansion of macro &#x27;pr_info&#x27;
 5241 |         pr_info(&quot;%s 1 point id=%d, fancurve_attr_id id=%d, fancurve.fan_speed_unit=%d, fancurve.size=%ld&quot;,
      |         ^~~~~~~
/build/source/kernel_module/legion-laptop.c:5233:25: note: &#x27;fancurve&#x27; declared here
 5233 |         struct fancurve fancurve;
      |                         ^~~~~~~~
cc1: some warnings being treated as errors
make[2]: *** [/nix/store/6516wznka0xwdkzx0cpgfl4wmx8d5r5y-linux-5.15.194-dev/lib/modules/5.15.194/source/scripts/Makefile.build:289: /build/source/kernel_module/legion-laptop.o] Error 1
make[1]: *** [/nix/store/6516wznka0xwdkzx0cpgfl4wmx8d5r5y-linux-5.15.194-dev/lib/modules/5.15.194/source/Makefile:1926: /build/source/kernel_module] Error 2
make[1]: Leaving directory &#x27;/nix/store/6516wznka0xwdkzx0cpgfl4wmx8d5r5y-linux-5.15.194-dev/lib/modules/5.15.194/build&#x27;
make: *** [Makefile:13: all] Error 2</pre>
</details>
<details>
<summary>linuxKernel.packages.linux_6_1.lenovo-legion-module</summary>
<pre>no configure script, doing nothing
Running phase: buildPhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;buildPhase&quot; }
build flags: SHELL=/nix/store/pq80mryqppk9a2z0i36hg112pbvfjr2p-bash-5.3p3/bin/bash CC=/nix/store/hd13cxx232pfsxd2gp3c6jfdbyz07wn8-gcc-14.3.0/bin/gcc LD=/nix/store/7s6vdx40nsh9vm0n1hd27sd8shs9jkp4-binutils-2.44/bin/ld AR=/nix/store/frxwlync78r3l94863r2cbnz7ngy273j-gcc-wrapper-14.3.0/bin/ar NM=/nix/store/frxwlync78r3l94863r2cbnz7ngy273j-gcc-wrapper-14.3.0/bin/nm STRIP=/nix/store/7s6vdx40nsh9vm0n1hd27sd8shs9jkp4-binutils-2.44/bin/strip OBJCOPY=/nix/store/frxwlync78r3l94863r2cbnz7ngy273j-gcc-wrapper-14.3.0/bin/objcopy OBJDUMP=/nix/store/frxwlync78r3l94863r2cbnz7ngy273j-gcc-wrapper-14.3.0/bin/objdump READELF=/nix/store/frxwlync78r3l94863r2cbnz7ngy273j-gcc-wrapper-14.3.0/bin/readelf HOSTCC=/nix/store/frxwlync78r3l94863r2cbnz7ngy273j-gcc-wrapper-14.3.0/bin/cc HOSTCXX=/nix/store/frxwlync78r3l94863r2cbnz7ngy273j-gcc-wrapper-14.3.0/bin/c++ HOSTAR=/nix/store/gg384szlprd0pqw9xafjl98q750xi00n-binutils-wrapper-2.44/bin/ar HOSTLD=/nix/store/gg384szlprd0pqw9xafjl98q750xi00n-binutils-wrapper-2.44/bin/ld ARCH=arm64 CROSS_COMPILE= KBUILD_OUTPUT=/nix/store/d6v3wif6fq20pny2w5657a7xpf0dzpi9-linux-6.1.156-dev/lib/modules/6.1.156/build SHELL=bash KERNELVERSION=6.1.156 KSRC=/nix/store/d6v3wif6fq20pny2w5657a7xpf0dzpi9-linux-6.1.156-dev/lib/modules/6.1.156/build INSTALLDIR=/nix/store/n837r66q9xqgpk0fx4z635wlpb4b40p0-lenovo-legion-module-0.0.20-unstable-2025-07-11/lib/modules/6.1.156/kernel/drivers/platform/x86 MODDESTDIR=/nix/store/n837r66q9xqgpk0fx4z635wlpb4b40p0-lenovo-legion-module-0.0.20-unstable-2025-07-11/lib/modules/6.1.156/kernel/drivers/platform/x86 DKMSDIR=/nix/store/n837r66q9xqgpk0fx4z635wlpb4b40p0-lenovo-legion-module-0.0.20-unstable-2025-07-11/lib/modules/6.1.156/misc
make -C /nix/store/d6v3wif6fq20pny2w5657a7xpf0dzpi9-linux-6.1.156-dev/lib/modules/6.1.156/build M=/build/source/kernel_module modules
make[1]: Entering directory &#x27;/nix/store/d6v3wif6fq20pny2w5657a7xpf0dzpi9-linux-6.1.156-dev/lib/modules/6.1.156/build&#x27;
  CC [M]  /build/source/kernel_module/legion-laptop.o
/build/source/kernel_module/legion-laptop.c: In function &#x27;wmi_exec_ints&#x27;:
/build/source/kernel_module/legion-laptop.c:1504:18: error: implicit declaration of function &#x27;wmi_evaluate_method&#x27;; did you mean &#x27;wmidev_evaluate_method&#x27;? [8;;[https://gcc.gnu.org/onlinedocs/gcc-14.3.0/gcc/Warning-Options.html#index-Wimplicit-function-declaration-Wimplicit-function-declaration8;;]](https://gcc.gnu.org/onlinedocs/gcc-14.3.0/gcc/Warning-Options.html#index-Wimplicit-function-declaration%07-Wimplicit-function-declaration8;;%07])
 1504 |         status = wmi_evaluate_method(guid, instance, method_id, params,
      |                  ^~~~~~~~~~~~~~~~~~~
      |                  wmidev_evaluate_method
/build/source/kernel_module/legion-laptop.c: At top level:
/build/source/kernel_module/legion-laptop.c:1856:16: warning: &#x27;ecram_memoryio_write&#x27; defined but not used [8;;[https://gcc.gnu.org/onlinedocs/gcc-14.3.0/gcc/Warning-Options.html#index-Wno-unused-function-Wunused-function8;;]](https://gcc.gnu.org/onlinedocs/gcc-14.3.0/gcc/Warning-Options.html#index-Wno-unused-function%07-Wunused-function8;;%07])
 1856 | static ssize_t ecram_memoryio_write(const struct ecram_memoryio *ec_memoryio,
      |                ^~~~~~~~~~~~~~~~~~~~
make[2]: *** [/nix/store/d6v3wif6fq20pny2w5657a7xpf0dzpi9-linux-6.1.156-dev/lib/modules/6.1.156/source/scripts/Makefile.build:250: /build/source/kernel_module/legion-laptop.o] Error 1
make[1]: *** [/nix/store/d6v3wif6fq20pny2w5657a7xpf0dzpi9-linux-6.1.156-dev/lib/modules/6.1.156/source/Makefile:2025: /build/source/kernel_module] Error 2
make[1]: Leaving directory &#x27;/nix/store/d6v3wif6fq20pny2w5657a7xpf0dzpi9-linux-6.1.156-dev/lib/modules/6.1.156/build&#x27;
make: *** [Makefile:13: all] Error 2</pre>
</details>
<details>
<summary>linuxPackages.lenovo-legion-module</summary>
<pre>Running phase: buildPhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;buildPhase&quot; }
build flags: SHELL=/nix/store/pq80mryqppk9a2z0i36hg112pbvfjr2p-bash-5.3p3/bin/bash CC=/nix/store/hd13cxx232pfsxd2gp3c6jfdbyz07wn8-gcc-14.3.0/bin/gcc LD=/nix/store/7s6vdx40nsh9vm0n1hd27sd8shs9jkp4-binutils-2.44/bin/ld AR=/nix/store/frxwlync78r3l94863r2cbnz7ngy273j-gcc-wrapper-14.3.0/bin/ar NM=/nix/store/frxwlync78r3l94863r2cbnz7ngy273j-gcc-wrapper-14.3.0/bin/nm STRIP=/nix/store/7s6vdx40nsh9vm0n1hd27sd8shs9jkp4-binutils-2.44/bin/strip OBJCOPY=/nix/store/frxwlync78r3l94863r2cbnz7ngy273j-gcc-wrapper-14.3.0/bin/objcopy OBJDUMP=/nix/store/frxwlync78r3l94863r2cbnz7ngy273j-gcc-wrapper-14.3.0/bin/objdump READELF=/nix/store/frxwlync78r3l94863r2cbnz7ngy273j-gcc-wrapper-14.3.0/bin/readelf HOSTCC=/nix/store/frxwlync78r3l94863r2cbnz7ngy273j-gcc-wrapper-14.3.0/bin/cc HOSTCXX=/nix/store/frxwlync78r3l94863r2cbnz7ngy273j-gcc-wrapper-14.3.0/bin/c++ HOSTAR=/nix/store/gg384szlprd0pqw9xafjl98q750xi00n-binutils-wrapper-2.44/bin/ar HOSTLD=/nix/store/gg384szlprd0pqw9xafjl98q750xi00n-binutils-wrapper-2.44/bin/ld ARCH=arm64 CROSS_COMPILE= KBUILD_OUTPUT=/nix/store/28rcas59mv3hlqfqikcvvnm4pzyh7v4z-linux-6.12.53-dev/lib/modules/6.12.53/build SHELL=bash KERNELVERSION=6.12.53 KSRC=/nix/store/28rcas59mv3hlqfqikcvvnm4pzyh7v4z-linux-6.12.53-dev/lib/modules/6.12.53/build INSTALLDIR=/nix/store/jb2lkpi8xi490mzir07jxzbbpri0qjn7-lenovo-legion-module-0.0.20-unstable-2025-07-11/lib/modules/6.12.53/kernel/drivers/platform/x86 MODDESTDIR=/nix/store/jb2lkpi8xi490mzir07jxzbbpri0qjn7-lenovo-legion-module-0.0.20-unstable-2025-07-11/lib/modules/6.12.53/kernel/drivers/platform/x86 DKMSDIR=/nix/store/jb2lkpi8xi490mzir07jxzbbpri0qjn7-lenovo-legion-module-0.0.20-unstable-2025-07-11/lib/modules/6.12.53/misc
make -C /nix/store/28rcas59mv3hlqfqikcvvnm4pzyh7v4z-linux-6.12.53-dev/lib/modules/6.12.53/build M=/build/source/kernel_module modules
make[1]: Entering directory &#x27;/nix/store/28rcas59mv3hlqfqikcvvnm4pzyh7v4z-linux-6.12.53-dev/lib/modules/6.12.53/build&#x27;
  CC [M]  /build/source/kernel_module/legion-laptop.o
/build/source/kernel_module/legion-laptop.c: In function &#x27;wmi_exec_ints&#x27;:
/build/source/kernel_module/legion-laptop.c:1504:18: error: implicit declaration of function &#x27;wmi_evaluate_method&#x27;; did you mean &#x27;wmidev_evaluate_method&#x27;? [8;;[https://gcc.gnu.org/onlinedocs/gcc-14.3.0/gcc/Warning-Options.html#index-Wimplicit-function-declaration-Wimplicit-function-declaration8;;]](https://gcc.gnu.org/onlinedocs/gcc-14.3.0/gcc/Warning-Options.html#index-Wimplicit-function-declaration%07-Wimplicit-function-declaration8;;%07])
 1504 |         status = wmi_evaluate_method(guid, instance, method_id, params,
      |                  ^~~~~~~~~~~~~~~~~~~
      |                  wmidev_evaluate_method
/build/source/kernel_module/legion-laptop.c: At top level:
/build/source/kernel_module/legion-laptop.c:1856:16: warning: &#x27;ecram_memoryio_write&#x27; defined but not used [8;;[https://gcc.gnu.org/onlinedocs/gcc-14.3.0/gcc/Warning-Options.html#index-Wno-unused-function-Wunused-function8;;]](https://gcc.gnu.org/onlinedocs/gcc-14.3.0/gcc/Warning-Options.html#index-Wno-unused-function%07-Wunused-function8;;%07])
 1856 | static ssize_t ecram_memoryio_write(const struct ecram_memoryio *ec_memoryio,
      |                ^~~~~~~~~~~~~~~~~~~~
make[3]: *** [/nix/store/28rcas59mv3hlqfqikcvvnm4pzyh7v4z-linux-6.12.53-dev/lib/modules/6.12.53/source/scripts/Makefile.build:229: /build/source/kernel_module/legion-laptop.o] Error 1
make[2]: *** [/nix/store/28rcas59mv3hlqfqikcvvnm4pzyh7v4z-linux-6.12.53-dev/lib/modules/6.12.53/source/Makefile:1945: /build/source/kernel_module] Error 2
make[1]: *** [/nix/store/28rcas59mv3hlqfqikcvvnm4pzyh7v4z-linux-6.12.53-dev/lib/modules/6.12.53/source/Makefile:224: __sub-make] Error 2
make[1]: Leaving directory &#x27;/nix/store/28rcas59mv3hlqfqikcvvnm4pzyh7v4z-linux-6.12.53-dev/lib/modules/6.12.53/build&#x27;
make: *** [Makefile:13: all] Error 2</pre>
</details>
<details>
<summary>linuxKernel.packages.linux_hardened.lenovo-legion-module</summary>
<pre>Running phase: buildPhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;buildPhase&quot; }
build flags: SHELL=/nix/store/pq80mryqppk9a2z0i36hg112pbvfjr2p-bash-5.3p3/bin/bash CC=/nix/store/hd13cxx232pfsxd2gp3c6jfdbyz07wn8-gcc-14.3.0/bin/gcc LD=/nix/store/7s6vdx40nsh9vm0n1hd27sd8shs9jkp4-binutils-2.44/bin/ld AR=/nix/store/frxwlync78r3l94863r2cbnz7ngy273j-gcc-wrapper-14.3.0/bin/ar NM=/nix/store/frxwlync78r3l94863r2cbnz7ngy273j-gcc-wrapper-14.3.0/bin/nm STRIP=/nix/store/7s6vdx40nsh9vm0n1hd27sd8shs9jkp4-binutils-2.44/bin/strip OBJCOPY=/nix/store/frxwlync78r3l94863r2cbnz7ngy273j-gcc-wrapper-14.3.0/bin/objcopy OBJDUMP=/nix/store/frxwlync78r3l94863r2cbnz7ngy273j-gcc-wrapper-14.3.0/bin/objdump READELF=/nix/store/frxwlync78r3l94863r2cbnz7ngy273j-gcc-wrapper-14.3.0/bin/readelf HOSTCC=/nix/store/frxwlync78r3l94863r2cbnz7ngy273j-gcc-wrapper-14.3.0/bin/cc HOSTCXX=/nix/store/frxwlync78r3l94863r2cbnz7ngy273j-gcc-wrapper-14.3.0/bin/c++ HOSTAR=/nix/store/gg384szlprd0pqw9xafjl98q750xi00n-binutils-wrapper-2.44/bin/ar HOSTLD=/nix/store/gg384szlprd0pqw9xafjl98q750xi00n-binutils-wrapper-2.44/bin/ld ARCH=arm64 CROSS_COMPILE= KBUILD_OUTPUT=/nix/store/6bxg8hrkb5i57rcvbikkx4g89rlaip3h-linux-hardened-6.12.43-dev/lib/modules/6.12.43-hardened1/build SHELL=bash KERNELVERSION=6.12.43-hardened1 KSRC=/nix/store/6bxg8hrkb5i57rcvbikkx4g89rlaip3h-linux-hardened-6.12.43-dev/lib/modules/6.12.43-hardened1/build INSTALLDIR=/nix/store/h4n9025izy4z929dshlprdjg4dvn5dn8-lenovo-legion-module-0.0.20-unstable-2025-07-11/lib/modules/6.12.43-hardened1/kernel/drivers/platform/x86 MODDESTDIR=/nix/store/h4n9025izy4z929dshlprdjg4dvn5dn8-lenovo-legion-module-0.0.20-unstable-2025-07-11/lib/modules/6.12.43-hardened1/kernel/drivers/platform/x86 DKMSDIR=/nix/store/h4n9025izy4z929dshlprdjg4dvn5dn8-lenovo-legion-module-0.0.20-unstable-2025-07-11/lib/modules/6.12.43-hardened1/misc
make -C /nix/store/6bxg8hrkb5i57rcvbikkx4g89rlaip3h-linux-hardened-6.12.43-dev/lib/modules/6.12.43-hardened1/build M=/build/source/kernel_module modules
make[1]: Entering directory &#x27;/nix/store/6bxg8hrkb5i57rcvbikkx4g89rlaip3h-linux-hardened-6.12.43-dev/lib/modules/6.12.43-hardened1/build&#x27;
  CC [M]  /build/source/kernel_module/legion-laptop.o
/build/source/kernel_module/legion-laptop.c: In function &#x27;wmi_exec_ints&#x27;:
/build/source/kernel_module/legion-laptop.c:1504:18: error: implicit declaration of function &#x27;wmi_evaluate_method&#x27;; did you mean &#x27;wmidev_evaluate_method&#x27;? [8;;[https://gcc.gnu.org/onlinedocs/gcc-14.3.0/gcc/Warning-Options.html#index-Wimplicit-function-declaration-Wimplicit-function-declaration8;;]](https://gcc.gnu.org/onlinedocs/gcc-14.3.0/gcc/Warning-Options.html#index-Wimplicit-function-declaration%07-Wimplicit-function-declaration8;;%07])
 1504 |         status = wmi_evaluate_method(guid, instance, method_id, params,
      |                  ^~~~~~~~~~~~~~~~~~~
      |                  wmidev_evaluate_method
/build/source/kernel_module/legion-laptop.c: At top level:
/build/source/kernel_module/legion-laptop.c:1856:16: warning: &#x27;ecram_memoryio_write&#x27; defined but not used [8;;[https://gcc.gnu.org/onlinedocs/gcc-14.3.0/gcc/Warning-Options.html#index-Wno-unused-function-Wunused-function8;;]](https://gcc.gnu.org/onlinedocs/gcc-14.3.0/gcc/Warning-Options.html#index-Wno-unused-function%07-Wunused-function8;;%07])
 1856 | static ssize_t ecram_memoryio_write(const struct ecram_memoryio *ec_memoryio,
      |                ^~~~~~~~~~~~~~~~~~~~
make[3]: *** [/nix/store/6bxg8hrkb5i57rcvbikkx4g89rlaip3h-linux-hardened-6.12.43-dev/lib/modules/6.12.43-hardened1/source/scripts/Makefile.build:229: /build/source/kernel_module/legion-laptop.o] Error 1
make[2]: *** [/nix/store/6bxg8hrkb5i57rcvbikkx4g89rlaip3h-linux-hardened-6.12.43-dev/lib/modules/6.12.43-hardened1/source/Makefile:1945: /build/source/kernel_module] Error 2
make[1]: *** [/nix/store/6bxg8hrkb5i57rcvbikkx4g89rlaip3h-linux-hardened-6.12.43-dev/lib/modules/6.12.43-hardened1/source/Makefile:224: __sub-make] Error 2
make[1]: Leaving directory &#x27;/nix/store/6bxg8hrkb5i57rcvbikkx4g89rlaip3h-linux-hardened-6.12.43-dev/lib/modules/6.12.43-hardened1/build&#x27;
make: *** [Makefile:13: all] Error 2</pre>
</details>
<details>
<summary>linuxKernel.packages.linux_6_16.lenovo-legion-module</summary>
<pre>legion-laptop.c: In function &#x27;wmi_exec_ints&#x27;:
legion-laptop.c:1504:18: error: implicit declaration of function &#x27;wmi_evaluate_method&#x27;; did you mean &#x27;wmidev_evaluate_method&#x27;? [8;;[https://gcc.gnu.org/onlinedocs/gcc-14.3.0/gcc/Warning-Options.html#index-Wimplicit-function-declaration-Wimplicit-function-declaration8;;]](https://gcc.gnu.org/onlinedocs/gcc-14.3.0/gcc/Warning-Options.html#index-Wimplicit-function-declaration%07-Wimplicit-function-declaration8;;%07])
 1504 |         status = wmi_evaluate_method(guid, instance, method_id, params,
      |                  ^~~~~~~~~~~~~~~~~~~
      |                  wmidev_evaluate_method
legion-laptop.c: In function &#x27;legion_platform_profile_init&#x27;:
legion-laptop.c:5047:13: warning: unused variable &#x27;err&#x27; [8;;[https://gcc.gnu.org/onlinedocs/gcc-14.3.0/gcc/Warning-Options.html#index-Wno-unused-variable-Wunused-variable8;;]](https://gcc.gnu.org/onlinedocs/gcc-14.3.0/gcc/Warning-Options.html#index-Wno-unused-variable%07-Wunused-variable8;;%07])
 5047 |         int err;
      |             ^~~
legion-laptop.c: At top level:
legion-laptop.c:1856:16: warning: &#x27;ecram_memoryio_write&#x27; defined but not used [8;;[https://gcc.gnu.org/onlinedocs/gcc-14.3.0/gcc/Warning-Options.html#index-Wno-unused-function-Wunused-function8;;]](https://gcc.gnu.org/onlinedocs/gcc-14.3.0/gcc/Warning-Options.html#index-Wno-unused-function%07-Wunused-function8;;%07])
 1856 | static ssize_t ecram_memoryio_write(const struct ecram_memoryio *ec_memoryio,
      |                ^~~~~~~~~~~~~~~~~~~~
make[4]: *** [/nix/store/pcc8slv6dwwhsd229rjq3isr1i1mwl38-linux-6.16.12-dev/lib/modules/6.16.12/source/scripts/Makefile.build:287: legion-laptop.o] Error 1
make[3]: *** [/nix/store/pcc8slv6dwwhsd229rjq3isr1i1mwl38-linux-6.16.12-dev/lib/modules/6.16.12/source/Makefile:2003: .] Error 2
make[2]: *** [/nix/store/pcc8slv6dwwhsd229rjq3isr1i1mwl38-linux-6.16.12-dev/lib/modules/6.16.12/source/Makefile:248: __sub-make] Error 2
make[2]: Leaving directory &#x27;/build/source/kernel_module&#x27;
make[1]: *** [/nix/store/pcc8slv6dwwhsd229rjq3isr1i1mwl38-linux-6.16.12-dev/lib/modules/6.16.12/source/Makefile:248: __sub-make] Error 2
make[1]: Leaving directory &#x27;/nix/store/pcc8slv6dwwhsd229rjq3isr1i1mwl38-linux-6.16.12-dev/lib/modules/6.16.12/build&#x27;
make: *** [Makefile:13: all] Error 2</pre>
</details>
<details>
<summary>linuxPackages_latest.lenovo-legion-module</summary>
<pre>legion-laptop.c: In function &#x27;wmi_exec_ints&#x27;:
legion-laptop.c:1504:18: error: implicit declaration of function &#x27;wmi_evaluate_method&#x27;; did you mean &#x27;wmidev_evaluate_method&#x27;? [8;;[https://gcc.gnu.org/onlinedocs/gcc-14.3.0/gcc/Warning-Options.html#index-Wimplicit-function-declaration-Wimplicit-function-declaration8;;]](https://gcc.gnu.org/onlinedocs/gcc-14.3.0/gcc/Warning-Options.html#index-Wimplicit-function-declaration%07-Wimplicit-function-declaration8;;%07])
 1504 |         status = wmi_evaluate_method(guid, instance, method_id, params,
      |                  ^~~~~~~~~~~~~~~~~~~
      |                  wmidev_evaluate_method
legion-laptop.c: In function &#x27;legion_platform_profile_init&#x27;:
legion-laptop.c:5047:13: warning: unused variable &#x27;err&#x27; [8;;[https://gcc.gnu.org/onlinedocs/gcc-14.3.0/gcc/Warning-Options.html#index-Wno-unused-variable-Wunused-variable8;;]](https://gcc.gnu.org/onlinedocs/gcc-14.3.0/gcc/Warning-Options.html#index-Wno-unused-variable%07-Wunused-variable8;;%07])
 5047 |         int err;
      |             ^~~
legion-laptop.c: At top level:
legion-laptop.c:1856:16: warning: &#x27;ecram_memoryio_write&#x27; defined but not used [8;;[https://gcc.gnu.org/onlinedocs/gcc-14.3.0/gcc/Warning-Options.html#index-Wno-unused-function-Wunused-function8;;]](https://gcc.gnu.org/onlinedocs/gcc-14.3.0/gcc/Warning-Options.html#index-Wno-unused-function%07-Wunused-function8;;%07])
 1856 | static ssize_t ecram_memoryio_write(const struct ecram_memoryio *ec_memoryio,
      |                ^~~~~~~~~~~~~~~~~~~~
make[4]: *** [/nix/store/dlpkrzv4ic9ciqi4j3h3wfwpxfxah537-linux-6.17.3-dev/lib/modules/6.17.3/source/scripts/Makefile.build:287: legion-laptop.o] Error 1
make[3]: *** [/nix/store/dlpkrzv4ic9ciqi4j3h3wfwpxfxah537-linux-6.17.3-dev/lib/modules/6.17.3/source/Makefile:2011: .] Error 2
make[2]: *** [/nix/store/dlpkrzv4ic9ciqi4j3h3wfwpxfxah537-linux-6.17.3-dev/lib/modules/6.17.3/source/Makefile:248: __sub-make] Error 2
make[2]: Leaving directory &#x27;/build/source/kernel_module&#x27;
make[1]: *** [/nix/store/dlpkrzv4ic9ciqi4j3h3wfwpxfxah537-linux-6.17.3-dev/lib/modules/6.17.3/source/Makefile:248: __sub-make] Error 2
make[1]: Leaving directory &#x27;/nix/store/dlpkrzv4ic9ciqi4j3h3wfwpxfxah537-linux-6.17.3-dev/lib/modules/6.17.3/build&#x27;
make: *** [Makefile:13: all] Error 2</pre>
</details>
<details>
<summary>linuxKernel.packages.linux_6_6.lenovo-legion-module</summary>
<pre>Running phase: buildPhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;buildPhase&quot; }
build flags: SHELL=/nix/store/pq80mryqppk9a2z0i36hg112pbvfjr2p-bash-5.3p3/bin/bash CC=/nix/store/hd13cxx232pfsxd2gp3c6jfdbyz07wn8-gcc-14.3.0/bin/gcc LD=/nix/store/7s6vdx40nsh9vm0n1hd27sd8shs9jkp4-binutils-2.44/bin/ld AR=/nix/store/frxwlync78r3l94863r2cbnz7ngy273j-gcc-wrapper-14.3.0/bin/ar NM=/nix/store/frxwlync78r3l94863r2cbnz7ngy273j-gcc-wrapper-14.3.0/bin/nm STRIP=/nix/store/7s6vdx40nsh9vm0n1hd27sd8shs9jkp4-binutils-2.44/bin/strip OBJCOPY=/nix/store/frxwlync78r3l94863r2cbnz7ngy273j-gcc-wrapper-14.3.0/bin/objcopy OBJDUMP=/nix/store/frxwlync78r3l94863r2cbnz7ngy273j-gcc-wrapper-14.3.0/bin/objdump READELF=/nix/store/frxwlync78r3l94863r2cbnz7ngy273j-gcc-wrapper-14.3.0/bin/readelf HOSTCC=/nix/store/frxwlync78r3l94863r2cbnz7ngy273j-gcc-wrapper-14.3.0/bin/cc HOSTCXX=/nix/store/frxwlync78r3l94863r2cbnz7ngy273j-gcc-wrapper-14.3.0/bin/c++ HOSTAR=/nix/store/gg384szlprd0pqw9xafjl98q750xi00n-binutils-wrapper-2.44/bin/ar HOSTLD=/nix/store/gg384szlprd0pqw9xafjl98q750xi00n-binutils-wrapper-2.44/bin/ld ARCH=arm64 CROSS_COMPILE= KBUILD_OUTPUT=/nix/store/pjrk1x15vdc6qhp1z0mam804qsaxi030-linux-6.6.112-dev/lib/modules/6.6.112/build SHELL=bash KERNELVERSION=6.6.112 KSRC=/nix/store/pjrk1x15vdc6qhp1z0mam804qsaxi030-linux-6.6.112-dev/lib/modules/6.6.112/build INSTALLDIR=/nix/store/kmxbajx3ywylcfvxj5qhpv59841yjia3-lenovo-legion-module-0.0.20-unstable-2025-07-11/lib/modules/6.6.112/kernel/drivers/platform/x86 MODDESTDIR=/nix/store/kmxbajx3ywylcfvxj5qhpv59841yjia3-lenovo-legion-module-0.0.20-unstable-2025-07-11/lib/modules/6.6.112/kernel/drivers/platform/x86 DKMSDIR=/nix/store/kmxbajx3ywylcfvxj5qhpv59841yjia3-lenovo-legion-module-0.0.20-unstable-2025-07-11/lib/modules/6.6.112/misc
make -C /nix/store/pjrk1x15vdc6qhp1z0mam804qsaxi030-linux-6.6.112-dev/lib/modules/6.6.112/build M=/build/source/kernel_module modules
make[1]: Entering directory &#x27;/nix/store/pjrk1x15vdc6qhp1z0mam804qsaxi030-linux-6.6.112-dev/lib/modules/6.6.112/build&#x27;
  CC [M]  /build/source/kernel_module/legion-laptop.o
/build/source/kernel_module/legion-laptop.c: In function &#x27;wmi_exec_ints&#x27;:
/build/source/kernel_module/legion-laptop.c:1504:18: error: implicit declaration of function &#x27;wmi_evaluate_method&#x27;; did you mean &#x27;wmidev_evaluate_method&#x27;? [8;;[https://gcc.gnu.org/onlinedocs/gcc-14.3.0/gcc/Warning-Options.html#index-Wimplicit-function-declaration-Wimplicit-function-declaration8;;]](https://gcc.gnu.org/onlinedocs/gcc-14.3.0/gcc/Warning-Options.html#index-Wimplicit-function-declaration%07-Wimplicit-function-declaration8;;%07])
 1504 |         status = wmi_evaluate_method(guid, instance, method_id, params,
      |                  ^~~~~~~~~~~~~~~~~~~
      |                  wmidev_evaluate_method
/build/source/kernel_module/legion-laptop.c: At top level:
/build/source/kernel_module/legion-laptop.c:1856:16: warning: &#x27;ecram_memoryio_write&#x27; defined but not used [8;;[https://gcc.gnu.org/onlinedocs/gcc-14.3.0/gcc/Warning-Options.html#index-Wno-unused-function-Wunused-function8;;]](https://gcc.gnu.org/onlinedocs/gcc-14.3.0/gcc/Warning-Options.html#index-Wno-unused-function%07-Wunused-function8;;%07])
 1856 | static ssize_t ecram_memoryio_write(const struct ecram_memoryio *ec_memoryio,
      |                ^~~~~~~~~~~~~~~~~~~~
make[3]: *** [/nix/store/pjrk1x15vdc6qhp1z0mam804qsaxi030-linux-6.6.112-dev/lib/modules/6.6.112/source/scripts/Makefile.build:243: /build/source/kernel_module/legion-laptop.o] Error 1
make[2]: *** [/nix/store/pjrk1x15vdc6qhp1z0mam804qsaxi030-linux-6.6.112-dev/lib/modules/6.6.112/source/Makefile:1924: /build/source/kernel_module] Error 2
make[1]: *** [/nix/store/pjrk1x15vdc6qhp1z0mam804qsaxi030-linux-6.6.112-dev/lib/modules/6.6.112/source/Makefile:234: __sub-make] Error 2
make[1]: Leaving directory &#x27;/nix/store/pjrk1x15vdc6qhp1z0mam804qsaxi030-linux-6.6.112-dev/lib/modules/6.6.112/build&#x27;
make: *** [Makefile:13: all] Error 2</pre>
</details>
</details>


---
### `x86_64-darwin`
:white_check_mark: *No rebuilds*


---
### `aarch64-darwin`
:white_check_mark: *No rebuilds*

--- a/pkgs/os-specific/linux/lenovo-legion/app.nix
+++ b/pkgs/os-specific/linux/lenovo-legion/app.nix
@@ -64,7 +64,7 @@ python3.pkgs.buildPythonApplication rec {
     description = "Utility to control Lenovo Legion laptop";
     homepage = "https://github.com/johnfanv2/LenovoLegionLinux";
     license = lib.licenses.gpl2Only;
-    platforms = lib.platforms.linux;
+    platforms = [ "x86_64-linux" ];
     maintainers = with lib.maintainers; [
       ulrikstrid
       logger

--- a/pkgs/os-specific/linux/lenovo-legion/default.nix
+++ b/pkgs/os-specific/linux/lenovo-legion/default.nix
@@ -34,7 +34,7 @@ stdenv.mkDerivation {
     description = "Linux kernel module for controlling fan and power in Lenovo Legion laptops";
     homepage = "https://github.com/johnfanv2/LenovoLegionLinux";
     license = lib.licenses.gpl2Only;
-    platforms = lib.platforms.linux;
+    platforms = [ "x86_64-linux" ];
     maintainers = [ lib.maintainers.ulrikstrid ];
     broken = kernel.kernelOlder "5.15";
   };

